packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ModelProvider.cs

```diff
@@ -109,15 +124,42 @@ private IReadOnlyList<ParameterProvider> BuildConstructorParameters()
                 {
                     if (!property.IsReadOnly)
                     {
-                        constructorParameters.Add(new ParameterProvider(property)
+                        var parameter = new ParameterProvider(property)
                         {
                             Type = propertyType.InputType
-                        });
+                        };
+                        if (!parameterNames.Contains(parameter.Name))
+                        {
+                            constructorParameters.Add(parameter);
+                        }
                     }
                 }
             }
 
-            return constructorParameters;
+            return (constructorParameters, constructorInitializer);
+
+            static ConstructorSignature? GetBaseConstructor(CSharpType? baseType)
+            {
+                // find the constructor on the base type
+                if (baseType is not { IsFrameworkType: false, Implementation: TypeProvider baseModel })
+                {
+                    return null;
+                }
+
+                if (baseModel.Constructors.Count == 0)
+                {
+                    return null;
+                }
+
+                // we cannot know which ctor to call, but in our implementation, there should only be one
+                var ctor = baseModel.Constructors[0];
```

Comment:
nit: maybe `var ctor = baseModel.Constructors.Single();` to ensure the single implementation.

Reply:
this part has been reworked